### PR TITLE
Introduce HelmValues in place of map[string]interface{}

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/helm/values.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type HelmValues map[string]interface{}
+
+func (h HelmValues) GetString(key string) (string, bool, error) {
+	return unstructured.NestedString(h, toKeys(key)...)
+}
+
+func (h HelmValues) Set(key string, val string) error {
```
Comment (jwendell) [inline pkg/helm/values.go]: Should this be called `SetString`?

File: pkg/helm/values.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type HelmValues map[string]interface{}
+
+func (h HelmValues) GetString(key string) (string, bool, error) {
```
Comment (jwendell) [inline pkg/helm/values.go:13]: Can we have comments for those exported functions?

File: pkg/helm/values.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package helm
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type HelmValues map[string]interface{}
```
Comment (jwendell) [inline pkg/helm/values.go]: Should we have methods for other types (int, struct, etc) as well?

File: pkg/helm/values.go (hunk 1 above)
Comment (luksa) [reply pkg/helm/values.go]: Ah. No, `Set` is fine, but the second argument should be `val interface{}`. Thanks!

File: pkg/helm/values.go (hunk 3 above)
Comment (luksa) [reply pkg/helm/values.go]: Not yet. Would be dead code.

File: pkg/helm/values.go (hunk 2 above)
Comment (luksa) [reply pkg/helm/values.go:13]: Done.

File: pkg/helm/values.go (hunk 1 above)
Comment (luksa) [reply pkg/helm/values.go]: Fixed.

**Review by jwendell — APPROVED** [review]
should we replace `interface{}` with `any` in the future?

**Comment by luksa** [comment]
> should we replace `interface{}` with `any` in the future?

No. Let's replace it now. :)  

Thanks!

File: pkg/helm/values.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+// GetString returns the string value of a nested field.
+// Returns false if value is not found and an error if not a string.
+func (h HelmValues) GetString(key string) (string, bool, error) {
+	return unstructured.NestedString(h, toKeys(key)...)
+}
+
+// Set sets the value of a nested field to a deep copy of the value provided.
+// Returns an error if value cannot be set because one of the nesting levels is not a map[string]any.
+func (h HelmValues) Set(key string, val interface{}) error {
```
Comment (jwendell) [inline pkg/helm/values.go]: still `interface{}`

Comment (luksa) [reply pkg/helm/values.go]: Thanks. I missed this one.

**Pull request merged**